packages/paima-sdk/paima-utils/src/config/loading.ts

```diff
@@ -0,0 +1,154 @@
+  chainCurrencyName: Type.String({ default: 'UNKNOWN_CURRENCY_NAME' }),
+  chainCurrencySymbol: Type.String({ default: 'NONAME' }),
+  chainCurrencyDecimals: Type.Number({ default: 0 }),
+  // TODO: this depends on the settings actually, but hard to express that
+  //ENV.BLOCK_TIME - 0.1
+  blockTime: Type.Number({ default: 4 }),
+  pollingRate: Type.Number({ default: 4 - 0.1 }),
+  deployment: Type.String({ default: '' }),
+  paimaL2ContractAddress: Type.RegExp(/^0x[0-9a-fA-F]{40}$/i),
+  funnelBlockGroupSize: Type.Number({ default: 100 }),
+});
+
+export const CardanoConfigSchema = Type.Object({
+  carpUrl: Type.String(),
+  network: Type.String(),
+  confirmationDepth: Type.Number(),
+});
+
+export type CardanoConfig = Static<typeof CardanoConfigSchema>;
+
+export const TaggedConfig = <T extends boolean>(T: T) =>
+  Type.Union([
+    Type.Intersect([
+      T ? EvmConfigSchema : Type.Partial(EvmConfigSchema),
+      Type.Required(
+        Type.Object({
+          type: Type.Union([
+            Type.Literal(ConfigNetworkType.EVM),
+            Type.Literal(ConfigNetworkType.EVM_OTHER),
+          ]),
+        })
+      ),
+    ]),
+    Type.Intersect([
+      CardanoConfigSchema,
+      Type.Object({ type: Type.Literal(ConfigNetworkType.CARDANO) }),
+    ]),
+  ]);
+
+export const BaseConfig = <T extends boolean>(T: T) => Type.Record(Type.String(), TaggedConfig(T));
+
+export const BaseConfigWithoutDefaults = BaseConfig(false);
+export const BaseConfigWithDefaults = BaseConfig(true);
+export const FullEvmConfig = EvmConfigSchema;
+
+const evmConfigDefaults = (blockTime: number | undefined) => ({
+  chainUri: '',
+  chainId: 0,
+  chainExplorerUri: '',
+  chainCurrencyName: 'UNKNOWN_CURRENCY_NAME',
+  chainCurrencySymbol: 'NONAME',
+  chainCurrencyDecimals: 0,
+  blockTime: 4,
+  pollingRate: (blockTime || 4) - 0.1,
+  deployment: '',
+  funnelBlockGroupSize: 100,
+});
+
+export async function loadConfig(): Promise<Static<typeof BaseConfigWithDefaults> | undefined> {
+  let configFileData: string;
+  try {
+    // TODO: would be nice to also read .yaml
+    configFileData = await fs.readFile(`config.${ENV.NETWORK}.yml`, 'utf8');
+  } catch (err) {
+    throw new Error('config file not found');
+  }
+
+  try {
+    const config = parseConfigFile(configFileData);
+
+    for (const network of Object.keys(config)) {
+      const networkConfig = config[network];
+
+      switch (networkConfig.type) {
+        case ConfigNetworkType.EVM_OTHER:
+        case ConfigNetworkType.EVM:
+          // TODO: current version of typebox doesn't have Value.Default
+          config[network] = Object.assign(
+            evmConfigDefaults(networkConfig.blockTime),
+            networkConfig
+          );
+          break;
+        case ConfigNetworkType.CARDANO:
+        default:
+          throw new Error('unknown network type');
+      }
+    }
+
+    return config;
+  } catch (err) {
+    // doLog(`Invalid config file: ${err}`);
```

Comment:
is this meant to be commented out?

Reply:
not really... But I have some weird issue with the build sometimes because of the `doLog` import, and the `ENV` import. But it doesn't happen all the time, and I couldn't find a way to fix it yet. For some reason it complains in paima-runtime about imports not being under the rootDir or something like that.

Reply:
Imports not being under the root dir is caused by the `import ... from .../src/...` issue I mentioned (since instead of importing a package, you're now importing a JS file that is under a totally different folder)